CMakeLists.txt

```diff
@@ -72,6 +72,8 @@ if(NOT MSVC)
    target_compile_options(snmalloc_lib INTERFACE -mcx16)
   elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "amd64")
    target_compile_options(snmalloc_lib INTERFACE -mcx16)
+  elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "AMD64")
```

Comment:
Can this not be a case-insensitive string compare (e.g. use TOLOWER on the processor and then compare against amd64)?